addons/l10n_ro_edi_stock_batch/models/stock_picking_batch.py

```diff
@@ -0,0 +1,120 @@
+
+        return records
+
+    @api.returns(None, lambda value: value[0])
+    def copy_data(self, default=None):
+        self.ensure_one()
+        if default is None:
+            default = {}
+
+        if self.l10n_ro_edi_stock_interface_id:
+            copy_data = self.l10n_ro_edi_stock_interface_id.copy_data()[0]
+            document_copy = self.env['l10n_ro.edi.stock.etransport.interface'].create(copy_data)
+            default['l10n_ro_edi_stock_interface_id'] = document_copy.id
+
+        return super().copy_data(default)
+
+    @api.onchange('l10n_ro_edi_stock_operation_type')
+    def _reset_location_type(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_type()
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_start_loc_type')
+    def _reset_start_location_data(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_data('start')
+
+    @api.onchange('l10n_ro_edi_stock_operation_type', 'l10n_ro_edi_stock_end_loc_type')
+    def _reset_end_location_data(self):
+        self.l10n_ro_edi_stock_interface_id._reset_location_data('end')
+
+    @api.depends('company_id.country_id')
+    def _compute_l10n_ro_edi_stock_enable(self):
+        for batch in self:
+            batch.l10n_ro_edi_stock_enable = batch.company_id.country_id.code == 'RO'
+
+    @api.depends('l10n_ro_edi_stock_enable', 'state', 'l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_send(self):
+        for batch in self:
+            batch.l10n_ro_edi_stock_enable_send = (batch.l10n_ro_edi_stock_enable
+                                                   and batch.state != 'draft'
+                                                   and batch.l10n_ro_edi_stock_state in (False, 'stock_sending_failed')
+                                                   and not batch.l10n_ro_edi_stock_interface_id._get_last_stock_validated_document())
+
+    @api.depends('l10n_ro_edi_stock_enable', 'state', 'l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_fetch(self):
+        for batch in self:
+            batch.l10n_ro_edi_stock_enable_fetch = batch.l10n_ro_edi_stock_enable and batch.l10n_ro_edi_stock_state == 'stock_sent'
+
+    @api.depends('l10n_ro_edi_stock_document_message')
+    def _compute_l10n_ro_edi_stock_enable_send_info(self):
+        for batch in self:
+            batch.l10n_ro_edi_stock_enable_send_info = batch.l10n_ro_edi_stock_document_message and True
+
+    @api.depends('l10n_ro_edi_stock_state')
+    def _compute_l10n_ro_edi_stock_enable_amend(self):
+        for batch in self:
+            batch.l10n_ro_edi_stock_enable_amend = (batch.l10n_ro_edi_stock_enable
+                                                    and batch.l10n_ro_edi_stock_state == 'stock_validated' or (batch.l10n_ro_edi_stock_state == 'stock_sending_failed' and batch.l10n_ro_edi_stock_interface_id._get_last_stock_validated_document()))
```

Comment:
```suggestion
            batch.l10n_ro_edi_stock_enable_amend = (batch.l10n_ro_edi_stock_enable
                                                    and batch.l10n_ro_edi_stock_state == 'stock_validated'
                                                    or (batch.l10n_ro_edi_stock_state == 'stock_sending_failed'
                                                        and batch.l10n_ro_edi_stock_interface_id._get_last_stock_validated_document()))
```
Don't forget to split your long lines :)